src/Uno.UI/UI/Xaml/Media/ImageSource.cs

```diff
@@ -88,6 +85,21 @@ protected ImageSource(Uri uri) : this()
 			InitFromUri(uri);
 		}
 
+		internal static Uri TryCreateUriFromString(string url)
+		{
+			if (url.StartsWith("/"))
+			{
+				url = MsAppXScheme + "://" + url;
+			}
+
+			if (url.HasValueTrimmed() && Uri.TryCreate(url.Trim(), UriKind.RelativeOrAbsolute, out var uri))
+			{
+				return uri;
+			}
+
+			return null;
```

Comment:
This logic will need a small adjustment to append `ms-appx:///` at the beginning if the url doesn't already have it and it doesn't start with `/`.